src/operations/operation.ts

```diff
@@ -67,6 +67,9 @@ export abstract class AbstractOperation<TResult = any> {
   /** @internal */
   timeoutMS?: number;
 
+  /** @internal */
+  timeoutContext!: TimeoutContext;
+
```

Comment:
NOTE: This is marked with a non-null assertion because the field will be set during execute_operation and only referenced within that function in the following places at which point, if it is not defined, this would indicate a bug in the driver. I can add an inline comment explaining this

1. `Topology.selectServer`
2. `Server.command`
3. `ConnectionPool.checkOut`
4. `Connection.command`